src/NServiceBus.Gateway.RavenDB/RavenGatewayDeduplicationConfiguration.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace NServiceBus.Gateway.RavenDB
+{
+    using ObjectBuilder;
+    using Settings;
+    using Raven.Client.Documents;
+    using Raven.Client.Documents.Operations.Expiration;
+    using Raven.Client.ServerWide.Commands;
+    using System;
+
+    /// <summary>
+    /// Configures the deduplication storage.
+    /// </summary>
+    public class RavenGatewayDeduplicationConfiguration : GatewayDeduplicationConfiguration
+    {
+        /// <summary>
+        /// Initialize a new instance of the RavenDB gateway deduplication configuration
+        /// </summary>
+        public RavenGatewayDeduplicationConfiguration(Func<IBuilder, ReadOnlySettings, IDocumentStore> documentStoreFactory)
+        {
+            Guard.AgainstNull(nameof(documentStoreFactory), documentStoreFactory);
+
+            this.documentStoreFactory = documentStoreFactory;
+        }
+
+        /// <summary>
+        /// Invoked when the endpoint configuration completed to initialize the storage or
+        ///  verify configuration before the endpoint starts.
+        /// </summary>
+        public override void Setup(ReadOnlySettings settings)
+        {
+            this.settings = settings;
+
+            base.Setup(settings);
+        }
+
+        /// <summary>
+        /// Creates an instance of the deduplication storage.
+        /// </summary>
+        public override IGatewayDeduplicationStorage CreateStorage(IBuilder builder)
+        {
+            var documentStore = documentStoreFactory(builder, settings);
+
+            EnsureClusterConfiguration(documentStore);
+            EnableExpirationFeature(documentStore, FrequencyToRunDeduplicationDataCleanup);
+
+            return new RavenGatewayDeduplicationStorage(documentStore, DeduplicationDataTimeToLive);
+        }
+
+        static void EnableExpirationFeature(IDocumentStore documentStore, long frequencyToRunDeduplicationDataCleanup)
+        {
+            documentStore.Maintenance.Send(new ConfigureExpirationOperation(new ExpirationConfiguration
+            {
+                Disabled = false,
+                DeleteFrequencyInSec = frequencyToRunDeduplicationDataCleanup
+            }));
```

Comment:
This becomes a database-wide setting?

Reply:
unfortunately yes, that's the reason `frequencyToRunDeduplicationDataCleanup` is exposed to users. If the Expiration feature is already enabled for other reasons we cannot change its settings.